data_steward/cdr_cleaner/cleaning_rules/fix_unmapped_survey_answers.py

```diff
@@ -0,0 +1,198 @@
+"""Original Issues: DC-1043, DC-1053 
+
+PPI answers are mapped to standard answer concepts in concept_relationship through 'Maps to 
+value' (it has been this case historically and it is still the case now), however, there are a 
+bunch of PPI answer concepts missing such relationships in concept_relationship. Interestingly, 
+the corresponding standard PPI answer concepts could be found through 'Maps to'.  This might have 
+been a bug in the vocabulary and they probably should’ve used 'Maps to value' for mapping Answer 
+concepts. 
+
+There are survey answers (value_source_concept_ids) in observation that are mapped to a '0' 
+value_as_concept_id. Those unmapped survey answers could be standard concepts, non-standard 
+concepts that could be mapped to a standard concept through Maps to in concept_relationship, 
+or deprecated concepts that do not map to anything. 
+
+For the standard answers, we could just use it as-is for populating value_as_concept_id. However, 
+among the non-standard concepts, not all of the mapped standard concepts are classified as 
+'Answer' and they could belong to other concept_classes. Below is a list of the concept_class_id 
+of the mapped concepts: 
+
+Context-dependent
+Answer
+Clinical Finding
+Question
+Unit
+Module
+
+Question or Module concept classes don't make sense so will get excluded. In conclusion, 
+
+1. For standard concepts --> Set value_as_concept_id to the source_value_concept_id. 
+
+2. For deprecated concepts --> Set value_as_concept_id to 0. 
+Actually we don't need to do anything for this case because visit_as_concept_id is already 0. 
+
+3, For non-standard concepts --> Set value_as_concept_id to standard concept ids mapped through 
+'Maps to' for the concept classes ( 'Answer', 'Context-dependent', 'Clinical Finding', 
+'Unit') only. """
+
+import logging
+
+# Third party imports
+from jinja2 import Environment
+
+# Project imports
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule, query_spec_list
+from constants.bq_utils import WRITE_TRUNCATE
+from common import OBSERVATION
+from utils.bq import JINJA_ENV as jinja_env
+
+LOGGER = logging.getLogger(__name__)
+
+JIRA_ISSUE_NUMBERS = ['DC1043', 'DC1053']
+
+# Query to create tables in sandbox with the rows that will be updated per cleaning rule
+SANDBOX_FIX_UNMAPPED_SURVEY_ANSWERS_QUERY = jinja_env.from_string("""
+CREATE OR REPLACE TABLE `{{project}}.{{sandbox_dataset}}.{{sandbox_table}}` AS
+SELECT
+  o.*,
+  COALESCE(cr.concept_id_2, 0) AS new_value_as_concept_id
+FROM
+  `{{project}}.{{dataset}}.observation` AS o
+JOIN
+  `{{project}}.{{dataset}}.concept_relationship` AS cr
+ON
+  o.value_source_concept_id = cr.concept_id_1 AND cr.relationship_id = 'Maps to'
+JOIN 
```

Comment:
I didn't realize you could do an `AND` in the `ON` statement, this is cool and good to know!